core/src/main/java/com/netflix/iceberg/DataFiles.java

```diff
@@ -136,13 +136,33 @@ public static DataFile fromInputFile(InputFile file, PartitionData partition, Me
         location, format, partition, file.getLength(), DEFAULT_BLOCK_SIZE, metrics);
   }
 
+  public static DataFile fromInputFile(InputFile file, PartitionData partition, Metrics metrics,
```

Comment:
In the `Writer`, I think it would be better to have a `fromEncryptedOutputFile` method instead of this one.